packages/addresses/src/__tests__/address-test.ts

```diff
@@ -91,9 +143,36 @@ describe('Base58EncodedAddress', () => {
         });
         it('fatals when trying to deserialize a byte buffer shorter than 32-bytes', () => {
             const tooShortBuffer = new Uint8Array(Array(31).fill(0));
-            expect(() => address.deserialize(tooShortBuffer)).toThrow();
+            expect(() => address.decode(tooShortBuffer)).toThrow();
+        });
+        it('memoizes getBase58Encoder and getBase58Decoder when called multiple times', async () => {
+            expect.assertions(2);
+
+            // reload the module to reset memoized state
+            let getAddressCodec: typeof import('../address').getAddressCodec;
+            await jest.isolateModulesAsync(async () => {
+                const base58ModulePromise =
+                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                    // @ts-ignore
+                    import('../address');
+                getAddressCodec = (await base58ModulePromise).getAddressCodec;
+            });
+
+            address = getAddressCodec!();
+            address.encode(
+                '4wBqpZM9xaSheZzJSMawUHDgZ7miWfSsxmfVF5jJpYP' as Base58EncodedAddress<'4wBqpZM9xaSheZzJSMawUKKwhdpChKbZ5eu5ky4Vigw'>
+            );
+
+            address = getAddressCodec!();
+            address.encode(
+                '4wBqpZM9xaSheZzJSMawUHDgZ7miWfSsxmfVF5jJpYP' as Base58EncodedAddress<'4wBqpZM9xaSheZzJSMawUKKwhdpChKbZ5eu5ky4Vigw'>
+            );
+
+            expect(jest.mocked(getBase58Encoder)).toHaveBeenCalledTimes(1);
+            expect(jest.mocked(getBase58Decoder)).toHaveBeenCalledTimes(1);
```

Comment:
* Why call `encode` at all in this test?
* Is it at all worth doing `expect(addressCodecA).toBe(addressCodecB)`?